ethers-core/src/types/transaction/eip2718.rs

```diff
@@ -398,6 +401,143 @@ impl From<&Transaction> for TypedTransaction {
     }
 }
 
+impl TypedTransaction {
+    #[allow(dead_code)]
+    fn as_legacy_ref(&self) -> Result<&TransactionRequest, TypedTransactionError> {
```

Comment:
these are all private, and marked as dead?

I think they should rather return a simple `Option<>`

Reply:
agree